pkg/api/routes.go

```diff
@@ -394,6 +396,32 @@ func (rh *RouteHandler) GetManifest(response http.ResponseWriter, request *http.
 		return
 	}
 
+	if rh.c.RepoDB != nil {
+		// check is image is a signature
+		isSignature, _, _, err := storage.CheckIsImageSignature(name, content, reference,
+			rh.c.StoreController)
+		if err != nil {
+			if errors.Is(err, zerr.ErrOrphanSignature) {
+				rh.c.Log.Warn().Err(err).Msg("image has signature format but it doesn't sign any image")
+			} else {
+				rh.c.Log.Error().Err(err).Msg("can't check if manifest is a signature or not")
+				response.WriteHeader(http.StatusInternalServerError)
+
+				return
+			}
+		}
+
+		if !isSignature {
```

Comment:
So basically we don't want to increment the download count if signature is being downloaded?

Reply:
In this implementation, the signatures are not stored same as the other manifest entries in RepoDB. So there isn't a download counter for signatures.